drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,403 @@
+		return ret;
+	}
+#endif
+
+	/* set sensor default pm and odr */
+	LOG_WRN("%s: pm: %d, odr: %d", dev->name, cfg->pm, cfg->odr);
+	lis2dux12_md_t mode = {
+		.odr = cfg->odr,
+		.fs = cfg->range,
+	};
+	ret = lis2dux12_mode_set(ctx, &mode);
+	if (ret < 0) {
+		LOG_ERR("%s: odr init error (12.5 Hz)", dev->name);
+		return ret;
+	}
+
+	/* set sensor default scale (used to convert sample values) */
+	LOG_WRN("%s: range is %d", dev->name, cfg->range);
+	ret = lis2dux12_set_range(dev, cfg->range);
+	if (ret < 0) {
+		LOG_ERR("%s: range init error %d", dev->name, cfg->range);
+		return ret;
+	}
+
+	return 0;
+}
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "LIS2DUX12 driver enabled without any devices"
+#endif
+
+/*
+ * Device creation macro, shared by LIS2DUX12_DEFINE_SPI() and
+ * LIS2DUX12_DEFINE_I2C().
+ */
+
+#define LIS2DUX12_DEVICE_INIT(inst)                                                                \
+	SENSOR_DEVICE_DT_INST_DEFINE(inst, lis2dux12_init, NULL, &lis2dux12_data_##inst,           \
+				     &lis2dux12_config_##inst, POST_KERNEL,                        \
+				     CONFIG_SENSOR_INIT_PRIORITY, &lis2dux12_driver_api);
+
+/*
+ * Instantiation macros used when a device is on a SPI bus.
+ */
+
+#ifdef CONFIG_LIS2DUX12_TRIGGER
+#define LIS2DUX12_CFG_IRQ(inst) .gpio_int = GPIO_DT_SPEC_INST_GET(inst, irq_gpios),
+#else
+#define LIS2DUX12_CFG_IRQ(inst)
+#endif /* CONFIG_LIS2DUX12_TRIGGER */
+
+#define LIS2DUX12_SPI_OPERATION                                                                    \
+	(SPI_WORD_SET(8) | SPI_OP_MODE_MASTER | SPI_MODE_CPOL | SPI_MODE_CPHA)
+
+#define LIS2DUX12_CONFIG_SPI(inst)                                                                 \
+	{                                                                                          \
+		.ctx =                                                                             \
+			{                                                                          \
+				.read_reg = (stmdev_read_ptr)stmemsc_spi_read,                     \
+				.write_reg = (stmdev_write_ptr)stmemsc_spi_write,                  \
+				.mdelay = (stmdev_mdelay_ptr)stmemsc_mdelay,                       \
+				.handle = (void *)&lis2dux12_config_##inst.stmemsc_cfg,            \
+			},                                                                         \
+		.stmemsc_cfg =                                                                     \
+			{                                                                          \
+				.spi = SPI_DT_SPEC_INST_GET(inst, LIS2DUX12_SPI_OPERATION, 0),     \
+			},                                                                         \
+		.range = DT_INST_PROP(inst, range), .pm = DT_INST_PROP(inst, power_mode),          \
+		.odr = DT_INST_PROP(inst, odr),                                                    \
+		IF_ENABLED(DT_INST_NODE_HAS_PROP(inst, irq_gpios), (LIS2DUX12_CFG_IRQ(inst)))      \
```

Comment:
here we should check int1_gpios and int2_gpios (see my comment for dts).
They can be put in or using UTIL_OR().